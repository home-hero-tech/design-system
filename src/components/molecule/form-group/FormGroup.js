import React from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';
import { Col } from 'react-flexbox-grid';

import css from './FormGroup.module.scss';

const FormGroup = ({ error, success, children, ...otherProps }) => {
  const classes = classNames({
    [css['c-form-group']]: true,
    [css['c-form-group--error']]: error,
    [css['c-form-group--success']]: success
  });

  return (
    <Col {...otherProps} className={classes}>
      {children}
    </Col>
  );
};

FormGroup.propTypes = {
  error: PropTypes.bool,
  success: PropTypes.bool,
  children: PropTypes.oneOfType([PropTypes.object, PropTypes.array])
};
FormGroup.defaultProps = {
  error: false,
  success: false,
  children: null
};

export default FormGroup;
