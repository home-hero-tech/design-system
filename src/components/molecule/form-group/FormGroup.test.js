import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import FormGroup from './FormGroup';

describe('FormGroup', () => {
  it('should match with the snapshot', () => {
    const wrapper = shallow(<FormGroup placeholder="placeholder text" id="ipt" name="ipt" type="text" label="Text" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should match with the snapshot with Message', () => {
    const wrapper = shallow(
      <FormGroup
        placeholder="placeholder text"
        id="ipt"
        name="ipt"
        type="text"
        label="Text"
        message="Default message"
      />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
