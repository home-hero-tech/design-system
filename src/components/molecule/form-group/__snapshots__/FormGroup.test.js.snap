// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormGroup should match with the snapshot 1`] = `
<Col
  className="c-form-group"
  id="ipt"
  label="Text"
  name="ipt"
  placeholder="placeholder text"
  type="text"
/>
`;

exports[`FormGroup should match with the snapshot with Message 1`] = `
<Col
  className="c-form-group"
  id="ipt"
  label="Text"
  message="Default message"
  name="ipt"
  placeholder="placeholder text"
  type="text"
/>
`;
